src/dodal/devices/detector/detector_motion.py

```diff
@@ -10,28 +13,34 @@ class ShutterState(IntEnum):
 
 
 class DetectorMotion(Device):
-    """Physical motion and interlocks for detector travel"""
-
-    upstream_x = Cpt(EpicsMotor, "-MO-DET-01:UPSTREAMX")
-    downstream_x = Cpt(EpicsMotor, "-MO-DET-01:DOWNSTREAMX")
-    x = Cpt(EpicsMotor, "-MO-DET-01:X")
-    y = Cpt(EpicsMotor, "-MO-DET-01:Y")
-    z = Cpt(EpicsMotor, "-MO-DET-01:Z")
-    yaw = Cpt(EpicsMotor, "-MO-DET-01:YAW")
-    shutter = Cpt(EpicsSignal, "-MO-DET-01:SET_SHUTTER_STATE")  # 0=closed, 1=open
-    #   monitors
-    shutter_closed_lim = Cpt(
-        EpicsSignalRO, "-MO-DET-01:CLOSE_LIMIT"
-    )  # on limit = 1, off = 0
-    shutter_open_lim = Cpt(
-        EpicsSignalRO, "-MO-DET-01:OPEN_LIMIT"
-    )  # on limit = 1, off = 0
-    z_disabled = Cpt(
-        EpicsSignalRO, "-MO-DET-01:Z:DISABLED"
-    )  # robot interlock, 0=ok to move, 1=blocked
-    crate_power = Cpt(
-        EpicsSignalRO, "-MO-PMAC-02:CRATE2_HEALTHY"
-    )  # returns 0 if no power
-    in_robot_load_safe_position = Cpt(
-        EpicsSignalRO, "-MO-PMAC-02:GPIO_INP_BITS.B2"
-    )  # returns 1 if safe
+    def __init__(self, prefix: str, name: str = ""):
+        device_prefix = "-MO-DET-01:"
+        self.upstream_x = Motor(f"{prefix}{device_prefix}UPSTREAMX")
+        self.downstream_x = Motor(f"{prefix}{device_prefix}DOWNSTREAMX")
+        self.x = Motor(f"{prefix}{device_prefix}X")
+        self.y = Motor(f"{prefix}{device_prefix}Y")
+        self.z = Motor(f"{prefix}{device_prefix}Z")
+        self.yaw = Motor(f"{prefix}{device_prefix}YAW")
+        self.shutter = Motor(f"{prefix}{device_prefix}SET_SHUTTER_STATE")
+
+        self.shutter = epics_signal_rw(
+            float, "-MO-DET-01:SET_SHUTTER_STATE"
+        )  # 0=closed, 1=open monitors
+
+        self.shutter_closed_lim = epics_signal_r(
+            float, f"{prefix}{device_prefix}CLOSE_LIMIT"
+        )  # on limit = 1, off = 0
+        self.shutter_open_lim = epics_signal_r(
+            float, f"{prefix}{device_prefix}OPEN_LIMIT"
+        )  # on limit = 1, off = 0
+        self.z_disabled = epics_signal_r(
+            float, f"{prefix}{device_prefix}Z:DISABLED"
+        )  # robot interlock, 0=ok to move, 1=blocked
+        self.crate_power = epics_signal_r(
+            float, f"{prefix}{device_prefix}CRATE2_HEALTHY"
+        )  # returns 0 if no power
+        self.in_robot_load_safe_position = epics_signal_r(
+            float, f"{prefix}{device_prefix}GPIO_INP_BITS.B2"
+        )  # returns 1 if safe
```

Comment:
Must: These are actually PVs on the `pmac` and so `dodal connect i03` is failing